test/extensions/filters/http/oauth2/oauth_integration_test.cc

```diff
@@ -91,8 +91,6 @@ TEST_F(OauthIntegrationTest, UnauthenticatedFlow) {
                                          {":authority", "authority"}};
   auto encoder_decoder = codec_client_->startRequest(headers);
 
-  Buffer::OwnedImpl buffer;
```

Comment:
I actually wonder if responding before the request is complete (since this is effectively what is tested now) is going to break some clients who don't handle this well? Unrelated to this test fix but maybe something to think about in the future. cc @rgs1.

Reply:
Shall we add the above as a note?

Reply:
Yeah it might be worth it to document this somewhere but up to you all as whether to do it in here or a follow up, etc.

Reply:
I think this is generic H2 behavior.  The proxy does have fully request headers for the GET by the time it replies.  I'll make a note to try to look at the more general issue in test frameworks.

Reply:
Sorry I know it's correct/generic behavior, I was just suggesting that I can envision some clients not dealing with this well so we should keep that in mind for future support queries.